preprocessing/__init__.py

```diff
@@ -2,36 +2,62 @@
 from color_refiment_helper import compare
 
 
-def check_graph_order(g: Graph, f: Graph):
+def checks(g, h) -> bool:
+    """
+    Collection method of all preprocessing checks
+
+    :param g: Graph
+    :param h: Graph
+    :return: Boolean: True if everything checks out
+    """
+    return check_degrees(g, h) and check_graph_size(g, h) and check_graph_order(g, h)
+
+
+def check_graph_order(g: Graph, h: Graph):
     """
     This method checks the orders (amount of vertices) of the graphs are the same
 
     :param g: Graph
-    :param f: Graph
+    :param h: Graph
     :return: Boolean: True if the amount of vertices are the same
     """
-    return len(g.vertices) == len(f.vertices)
+    return len(g.vertices) == len(h.vertices)
 
 
-def check_graph_size(g: Graph, f: Graph):
+def check_graph_size(g: Graph, h: Graph):
     """
     This method checks if the sizes (the amount of edges) of the graphs are the same
 
     :param g: Graph
-    :param f: Graph
+    :param h: Graph
     :return: Boolean: True if the amount of edges are the same
     """
-    return len(g.edges) == len(f.edges)
+    return len(g.edges) == len(h.edges)
 
 
-def check_degrees(g: Graph, f: Graph):
+def check_degrees(g: Graph, h: Graph):
     """
     This method checks if the degrees of all the vertices in the graphs are all the same
 
     :param g: Graph
-    :param f: Graph
+    :param h: Graph
     :return: Boolean: True if the degrees are the same
     """
     degree_g = [v.degree for v in g.vertices]
-    degree_f = [v.degree for v in f.vertices]
-    return compare(degree_g, degree_f)
+    degree_h = [v.degree for v in h.vertices]
+    return compare(degree_g, degree_h)
+
+
+def remove_loners(g: Graph):
+    """
+        Method removes
+        - Vertices with degree 0.
+        - corrollas & knots (vertices that have one or more loops and no non-loop edges)
+
+    :param g: Graph
+    :return: processed Graph g
+    """
+    for vertex in g.vertices:
+        if vertex.degree is 0 or (all(neigh == vertex for neigh in vertex.neighbours)):
```

Comment:
Kan dit in een simpele graaf (eg eentje zonder zelf-loops)?
Want volgens mij krijgen we alleen maar undirected simple graphs straks.

Reply:
heb je gelijk in, dat moet ik nog ff navragen, anders kan dit weg.